src/osd/OSDMap.cc

```diff
@@ -5987,7 +5987,12 @@ void OSDMap::check_health(CephContext *cct,
   }
 
   // OSD_UPGRADE_FINISHED
-  // none of these (yet) since we don't run until luminous upgrade is done.
+  if (auto require_release = pending_require_osd_release()) {
+    ostringstream ss;
+    ss << "all OSDs are running " << *require_release << " or later but"
+       << " require_osd_release < " << *require_release;
+    checks->add("OSD_UPGRADE_FINISHED", HEALTH_WARN, ss.str(), 0);
```

Comment:
could add the command for the user to run in the detail section

Reply:
I added the same string to the detail section as well.